qualtran/bloqs/swap_network/swap_with_zero.py

```diff
@@ -171,6 +180,19 @@ def wire_symbol(self, reg: Register, idx: Tuple[int, ...] = tuple()) -> 'WireSym
             return TextBox(f'swap{subscript}')
         raise ValueError(f'Unrecognized register name {name}')
 
+    def on_classical_vals(
+        self, *, targets: 'ClassicalValT', **selection: 'ClassicalValT'
+    ) -> Dict[str, 'ClassicalValT']:
+        assert isinstance(targets, np.ndarray)
+        selection_idx = tuple(selection.values())
+        for i, sel_idx_small, idx_one, idx_two in self._swap_sequence:
+            if selection_idx[i] & (1 << sel_idx_small) != 0:
```

Comment:
Add a note to help remind the reader why this bit pattern results in a swap :)

Reply:
This is just checking whether the `sel_idx_small`'th bit is 1 or 0 in the selection register. There is a detailed explanation in `_swap_with_zero_swap_sequence` which yields a sequence of swaps to enable the swap with zero.

Reply:
I know, I was hoping there was a TLDR for the _swap_with_zero_sequence

Reply:
Added a detailed docstring to `_swap_with_zero_swap_sequence`